cider/function/SubstraitFunctionCiderMappings.h

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef CIDER_FUNCTION_SUBSTRAITFUNCTIONMAPPINGS_H
+#define CIDER_FUNCTION_SUBSTRAITFUNCTIONMAPPINGS_H
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include "type/plan/Analyzer.h"
+#include "util/sqldefs.h"
+
+using FunctionSQLScalarOpsMappings = std::unordered_map<std::string, SQLOps>;
+using FunctionSQLAggOpsMappings = std::unordered_map<std::string, SQLAgg>;
+using FunctionSQLOpSupportTypeMappings =
+    std::unordered_map<std::string, OpSupportExprType>;
+
+class SubstraitFunctionCiderMappings {
+ public:
+  const SQLOps getFunctionScalarOp(const std::string& function_name) const {
+    const auto& scalar_op_map = scalarMappings();
+    auto iter = scalar_op_map.find(function_name);
+    if (iter == scalar_op_map.end()) {
+      return SQLOps::kUNDEFINED_OP;
+    }
+    return iter->second;
+  }
+
+  const SQLAgg getFunctionAggOp(const std::string& function_name) const {
+    const auto& agg_op_map = aggregateMappings();
+    auto iter = agg_op_map.find(function_name);
+    if (iter == agg_op_map.end()) {
+      return SQLAgg::kUNDEFINED_AGG;
+    }
+    return iter->second;
+  }
+
+  const OpSupportExprType getFunctionOpSupportType(
+      const std::string& function_name) const {
+    const auto& op_support_map = opsSupportTypeMappings();
+    auto iter = op_support_map.find(function_name);
+    if (iter == op_support_map.end()) {
+      return OpSupportExprType::kUNDEFINED_EXPR;
+    }
+    return iter->second;
+  }
+
+  /// scalar function names in difference between engine own and Substrait.
+  virtual const FunctionSQLScalarOpsMappings scalarMappings() const {
```

Comment:
why virtual?

Reply:
The code under cider/function/substrait is from velox, chaojun will separate these code from velox. So in the future the code will been readjusted. 
Now this virtual is used to inheritance VeloxToSubstraitFunctionMappings which mapping the velox's function name to substrait's function name. Maybe in the future there will been more inheritance mapping for deferent sql engine.